src/iexamine.cpp

```diff
@@ -73,7 +73,7 @@ void iexamine::gaspump(player *p, map *m, const tripoint &examp)
                 item spill( item_it->type->id, calendar::turn );
                 const auto min = item_it->liquid_charges( 1 );
                 ///\EFFECT_DEX decreases amount of gas spilled from a pump
-                const auto max = item_it->liquid_charges( 1 ) * 8.0 / p->dex_cur;
+                const auto max = item_it->liquid_charges( 1 ) * 8.0 / std::max( 1, p->dex_cur );
```

Comment:
This would work but there are several possible enhancements:

* You now can (and should where possible) pass `charges` to the `item` constructor.
* `max` is a confusing variable name and `auto` doesn't help either. Consider `qty` and `int`
* consider using `Character::get_dex()` accessor rather than the `dex_cur` member directly